pkg/terraform/exec/plugins/azureprivatedns/resource_private_dns_srv_record.go

```diff
@@ -174,15 +174,13 @@ func resourceArmPrivateDNSArvRecordDelete(d *schema.ResourceData, meta interface
 func flattenAzureRmPrivateDNSSrvRecords(records *[]privatedns.SrvRecord) []map[string]interface{} {
 	results := make([]map[string]interface{}, 0, len(*records))
 
-	if records != nil {
```

Comment:
Rather than getting rid of the check for nil, the check for nil should be done earlier in the function. The scan is telling you that, if `records` were nil, then the earlier dereference in the function would have thrown a panic before getting to this nil check.

Add the following at the top of the function.
```
if records == nil {
  return nil
}
```